src/indicators/drought/biomasse.py

```diff
@@ -0,0 +1,335 @@
+
+    Args:
+        admin: Admin level, one of 'ADM0', 'ADM1',
+            or 'ADM2'.
+        start_dekad: Starting dekad for annual
+            calculations.
+        reprocess: Boolean, whether to reprocess.
+        redownload: Boolean, whether to reprocess.
+        file_descriptor: If not None, file descriptor
+            to access non-admin file.
+
+    Returns:
+        Pandas data frame.
+    """
+    if reprocess:
+        return process_dmp(
+            admin=admin, start_dekad=start_dekad, redownload=redownload
+        )
+    else:
+        _check_admin(admin)
+        processed_path = _get_processed_path(
+            admin=admin, iso3=iso3, start_dekad=start_dekad
+        )
+        return pd.read_csv(processed_path)
+
+
+def aggregate_biomasse(
+    admin_pcodes: List[str],
+    admin: AdminArgument = "ADM2",
+    start_dekad: int = 10,
+    iso3: Union[bool, str] = None,
+):
+    """Aggregate biomasse data to set of areas
+
+    While the generic processed administrative
+    biomasse data is useful, sometimes we want
+    to aggregate the biomasse values to a set
+    of administrative areas. This should be done
+    by summing up the mean biomasse for each dekad
+    for those areas and the reported biomasse for
+    each year and dekad, then recalculating the
+    anomaly.
+    """
+    _check_admin(admin)
+    df = load_biomasse_data(admin=admin, start_dekad=start_dekad)
+    admin_col = admin.replace("ADM", "admin") + "Pcod"
+    df_subset = df[df[admin_col].isin(admin_pcodes)]
+    # just keep to year 2000 for unique value per dekad
+    # when calculating biomasse mean aggregated
+    df_mean = df_subset[df.year.isin([2000])]
+    df_mean = (
+        df_mean[["dekad", "biomasse_mean"]]
+        .groupby("dekad")
+        .sum()
+        .reset_index()
+    )
+    df_dekad = (
+        df_subset[["year", "dekad", "biomasse"]]
+        .groupby(["year", "dekad"])
+        .sum()
+        .reset_index()
+    )
+    df_merged = pd.merge(df_dekad, df_mean, on=["dekad"])
+    df_merged["biomasse_anomaly"] = (
+        100 * df_merged["biomasse"] / df_merged["biomasse_mean"]
+    )
+
+    # sorting data again year then dekad
+    df_merged.sort_values(by=["year", "dekad"], inplace=True)
+
+    if iso3 is not None:
+        processed_filepath = _get_processed_path(
+            iso3=iso3, admin=admin, start_dekad=start_dekad
+        )
+        df_merged.to_csv(processed_filepath, index=False)
+
+    return df_merged
+
+
+def load_aggregated_biomasse_data(file_descriptor: str, start_dekad: int = 10):
+    processed_filepath = _get_processed_path(
+        admin=file_descriptor, start_dekad=start_dekad
+    )
+    return pd.read_csv(processed_filepath)
+
+
+def load_biomasse_mean(mask: bool = True):
+    """Used to mask WRSI"""
+    bm_fp = os.path.join(_raw_path, "BiomassValueMean.tif")
+    da = rioxarray.open_rasterio(bm_fp)
+    if mask:
+        da = da.where(da.values <= 0, drop=True)
```

Comment:
I get an error when running `load_biomasse_mean()`. Seems to be caused by the `drop=True`. Assuming you didn't finalize this function yet but fyi

Reply:
Yes good catch!